### PR TITLE
release/1.5.0: site config & documentation updates round 3 (Gaea C4 and C5, Derecho, Parallelworks, Containers)

### PR DESCRIPTION
**Pull request by climbfuji**
### Summary

The title says it all: site config & documentation updates for Gaea C4 and C5, Derecho, Parallelworks, JCSDA CI containers for `release/1.5.0`.

Also updates the spack submodule pointer for the release branch for two bug fixes added during the installation on these systems (ESMF, openssl).

### Testing

See https://github.com/JCSDA/spack-stack/issues/765

### Applications affected

All applications on said systems.

### Systems affected

Gaea C4 and C5, Derecho, Parallelworks, JCSDA CI containers

### Dependencies

n/a

### Issue(s) addressed

Working towards https://github.com/JCSDA/spack-stack/issues/765

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [ ] ~~All dependency PRs/issues have been resolved and this PR can be merged.~~

### CODE REVIEW

File: configs/containers/README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ To avoid hardcoding specs in the generic container recipes, we keep the specs li
   specs: [base-env@1.0.0, jedi-base-env@1.0.0 ~fftw, ewok-env@1.0.0, jedi-fv3-env@1.0.0,
     jedi-mpas-env@1.0.0, bacio@2.4.1, bison@3.8.2, bufr@12.0.0, ecbuild@3.7.2, eccodes@2.27.0, ecflow@5,
     eckit@1.24.4, ecmwf-atlas@0.34.0 +trans ~fftw, fiat@1.2.0, ectrans@1.2.0 ~fftw, eigen@3.4.0,
-    fckit@0.11.0, fms@release-jcsda, g2@3.4.5, g2tmpl@1.10.2, gftl-shared@1.5.0,
+    fckit@0.11.0, fms@release-jcsda, g2@3.4.5, g2tmpl@1.10.0, gftl-shared@1.5.0,
```
Comment (climbfuji) [inline configs/containers/README.md:10]: Reverting, this is still not fixed.

Comment (climbfuji) [reply configs/containers/README.md:10]: I re-opened the corresponding issue so that we don't forget.

File: configs/sites/derecho/packages.yaml [hunk 2]
```diff
@@ -24,26 +24,16 @@ packages:
...
-    buildable: False
-    externals:
-    - spec: py-pip@22.0.4
-      prefix: /usr
 
 ### Modification of common packages
-  #esmf:
-  #  variants: ~xerces ~pnetcdf snapshot=none ~shared +external-parallelio esmf_os=Linux esmf_comm=mpich3
+  esmf:
+    variants: ~xerces ~pnetcdf snapshot=none ~shared +external-parallelio esmf_os=Linux esmf_comm=mpich
```
Comment (AlexanderRichert-NOAA) [inline configs/sites/derecho/packages.yaml:36]: For what it's worth, I _believe_ it is the case that you can just put the unique variants here (esmf_os, esmf_comm), but the rest of the ones from the common config should still be respected. That was my experience on Acorn, anyway.

**Review by AlexanderRichert-NOAA — APPROVED** [review]

**Review by ulmononian — APPROVED** [review]

File: configs/sites/derecho/packages.yaml (hunk 2 above)
Comment (climbfuji) [reply configs/sites/derecho/packages.yaml:36]: Ok, thanks. I will try this next time!

**Pull request merged**